src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+            }
+            return true;
+        }
+
+        private void SkipWhiteSpace()
+        {
+            //Create local copy to avoid bounds checks.
+            ReadOnlySpan<byte> localCopy = _buffer;
+            for (; _consumed < localCopy.Length; _consumed++)
+            {
+                byte val = localCopy[_consumed];
+                if (val != JsonConstants.Space &&
+                    val != JsonConstants.CarriageReturn &&
+                    val != JsonConstants.LineFeed &&
+                    val != JsonConstants.Tab)
+                {
+                    break;
+                }
+
+                if (val == JsonConstants.LineFeed)
+                {
+                    _lineNumber++;
+                    _lineBytePosition = 0;
+                }
+                else
+                {
+                    _lineBytePosition++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method contains the logic for processing the next value token and determining
+        /// what type of data it is.
+        /// </summary>
+        private bool ConsumeValue(byte marker)
+        {
+            if (marker == JsonConstants.Quote)
+                return ConsumeString();
+            else if (marker == JsonConstants.OpenBrace)
+                StartObject();
+            else if (marker == JsonConstants.OpenBracket)
+                StartArray();
+            else if ((uint)(marker - '0') <= '9' - '0' || marker == '-')
+                return ConsumeNumber();
+            else if (marker == 'f')
+                return ConsumeLiteral(JsonConstants.FalseValue, JsonTokenType.False);
+            else if (marker == 't')
+                return ConsumeLiteral(JsonConstants.TrueValue, JsonTokenType.True);
+            else if (marker == 'n')
+                return ConsumeLiteral(JsonConstants.NullValue, JsonTokenType.Null);
+            else
+            {
+                if (_readerOptions.CommentHandling != JsonCommentHandling.Default)
+                {
+                    if (_readerOptions.CommentHandling == JsonCommentHandling.AllowComments)
+                    {
+                        if (marker == JsonConstants.Solidus)
+                        {
+                            return ConsumeComment();
+                        }
+                    }
+                    else
+                    {
+                        // JsonCommentHandling.SkipComments
+                        if (marker == JsonConstants.Solidus)
+                        {
+                            if (SkipComment())
+                            {
+                                if (_consumed >= (uint)_buffer.Length)
+                                {
+                                    if (_isNotPrimitive && IsLastSpan)
+                                    {
+                                        if (_tokenType != JsonTokenType.EndArray && _tokenType != JsonTokenType.EndObject)
```

Comment:
Same question as earlier as to whether this can be combined into the previous if